install/first_google_boot.sh

```diff
@@ -222,17 +222,38 @@ function extract_spinnaker_kube_credentials() {
     gcloud container clusters get-credentials $kube_cluster --zone $kube_zone --project $kube_project
 
     if [[ -s $config_path ]]; then
-      echo "Kubernetes credentials successfully extracted to $config_path" 
+      echo "Kubernetes credentials successfully extracted to $config_path"
       chmod 400 $config_path
       chown spinnaker:spinnaker $config_path
+
+      write_default_value "SPINNAKER_KUBERNETES_ENABLED" "true"
     else
-      echo "Failed to extract kubernetes credentials to $config_path"  
+      echo "Failed to extract kubernetes credentials to $config_path"
       rm $config_path
     fi
-    write_default_value "SPINNAKER_KUBERNETES_ENABLED" "true"
+
+    # Safely return since we have attempted to configure Kubernetes with the
+    # provided metadata.
+    return
   else
     clear_instance_metadata "kube_cluster"
   fi
+
```

Comment:
Ah, I missed that return statement. That looks problematic. Minimally it would leave the other credentials hanging in the metadata. If you overspecify then you should know that. You are giving the cluster preference. Why would you do that since it is more work? Significantly more stepping back when you consider that you need additional scopes and are also forcing an upgrade to gcloud in order to use those APIs (which is affecting boot times -- we should move that to the bake).

What's the harm in giving precedence the other way around? You can interpret the cluster parameter as being available if needed and turns out that it isnt currently needed for anything else. So the implementation is superfluous but the interface is well defined and not overspecified (though could be surprising if the config was from a different cluster than the one you gave, but definitions still feel more robust).